### PR TITLE
[Enhancement] Added ARIA specification of First and Last option highlight

### PR DESCRIPTION
**Pull request by silviuavram**
Addresses https://github.com/paypal/downshift/issues/589

**What**:
Added highlight on first and last options when user opens the dropdown via Up/Down Arrows.

**Why**:
According to ARIA, if dropdown is collapsed and focus is on the text input:
- Arrow Down should open menu **and** highlight on option.
- Arrow Up should open menu **and** highlight last option.

**How**:

Modified `moveHighlightedIndex` in dropdown.js, specifically the `else` branch. There I open the menu, and then check if the opening action was made by Arrow up or down. I also check if there are any items in the dropdown.

If yes, I call `setHighlightedIndex` with either `itemCount - 1` or `0`, depending on the value of the event type (arrow up or arrow down), in order to comply to ARIA specifications.

I also modified some unit tests and end-to-end tests in order to reflect the changes. Also added 2 more end-to-end tests that test only this new behaviour.

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged
      
- [ ] Added myself to contributors table

### CODE REVIEW

File: src/__tests__/downshift.misc-with-utils-mocked.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ test('does not scroll from an onMouseMove event', () => {
   }
   const {queryByTestId} = render(<HighlightedIndexController />)
   const input = queryByTestId('input')
-  const item = queryByTestId('item-1')
+  const item = queryByTestId('item-2')
```
Comment (silviuavram) [inline src/__tests__/downshift.misc-with-utils-mocked.js]: I managed to 'fix' this test, although I need some more insight on what it tests. I used that example as a story and I still could not figure out how it works and how I managed to break it. Some help, please?

Comment (kentcdodds) [reply src/__tests__/downshift.misc-with-utils-mocked.js]: This is actually fine. Expected for the change that's been made.

File: src/downshift.js [hunk 2]
```diff
@@ -247,7 +247,24 @@ class Downshift extends Component {
     if (this.getState().isOpen) {
       this.changeHighlightedIndex(amount, otherStateToSet)
     } else {
-      this.setHighlightedIndex(undefined, {isOpen: true, ...otherStateToSet})
+      // if menu was closed, will open it.
```
Comment (kentcdodds) [inline src/downshift.js]: In this case we know it's closed because this is the else case of an isOpen check. We can remove the comment.

**Comment by kentcdodds** [comment]
Oh, and looks like some cypress tests are failing on CI: https://travis-ci.org/paypal/downshift/builds/437712230

**Comment by silviuavram** [comment]
Thank you for the quick feedback.

Those tests are the ones I added, will take a look and find out what the problem is.

Also will address the other comments and come back to you once the PR is polished.

**Comment by kentcdodds** [comment]
I'm going to manually merge this because I'm doing weird things. Standby...

**Comment by kentcdodds** [comment]
Sweet. Hopefully I can get this released in the major release this afternoon.

**Comment by kentcdodds** [comment]
Let's keep it open until it's actually released 👍

**Comment by kentcdodds** [comment]
This has been released in 3.0.0!

**Pull request merged**